### PR TITLE
add explanatory text of what to do on app version error

### PR DESCRIPTION
**Pull request by jancborchardt**
Please review @nickvergessen @rullzer @LukasReschke @MorrisJobke 

All warnings need an explanation of how to fix it. Otherwise people don’t know what to do and can’t fix the issue.

### CODE REVIEW

**Comment by jancborchardt** [comment]
Btw @nickvergessen this error shows up one by one, for every incompatible app. Instead of fixing all at once I need to fix everyone one by one and refresh. Can we show all the apps which don’t work in one, as we do elsewhere too?

**Comment by nickvergessen** [comment]
> this error shows up one by one, for every incompatible app. Instead of fixing all at once I need to fix everyone one by one and refresh. Can we show all the apps which don’t work in one, as we do elsewhere too?

Fixed

File: lib/base.php [hunk 1]
```diff
@@ -377,7 +377,7 @@ private static function printUpgradePage() {
 		\OCP\Util::addScript('update');
 		\OCP\Util::addStyle('update');
 
-		/** @var \OCP\App\IAppManager $appManager */
+		/** @var \OC\App\AppManager $appManager */
```
Comment (MorrisJobke) [inline lib/base.php]: Why?

**Review by MorrisJobke — CHANGES_REQUESTED** [review]

**Comment by jancborchardt** [comment]
@nickvergessen is it now the same block over again for every app? Can you make it this instead:

> The files of the following apps were not replaced correctly:
> - Activity (activity
> - Notifications (notifications)
> - PDF viewer (files_pdfviewer)
>
> Make sure the version is compatible with the server.

**Comment by nickvergessen** [comment]
No, it's a comma separated list:
> The files of the app Activity (activity), Deleted files (files_trashbin) were not replaced correctly. Make sure it is a version compatible with the server.


That has to be enough, because we're not going to add HTML support to HintExceptions, because that will have dangerous outcomes on other ends.

File: lib/base.php (hunk 1 above)
Comment (nickvergessen) [inline lib/base.php]: Because the used method below is not in the public interface

**Comment by jancborchardt** [comment]
@nickvergessen sounds good, thanks! :)

**Comment by jancborchardt** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**